### ISSUE TITLE
Add readonly modifier false positive for field of generic parameter type constrained to a struct

### ISSUE THREAD

**Issue by bartdesmet** [dotnet/roslyn#49290]
```csharp
interface I
{
    int P { get; set; }
}

class C<T> where T : struct, I
{
    private T _x; // IDE0044: Add readonly modifier

    public void Foo() => _x.P = 42;
}
```

Making the field `readonly` is invalid because `_x.P` mutates the value. Adding the `readonly` modifier yields:

```
CS1648	Members of readonly field 'C<T>._x' cannot be modified (except in a constructor or a variable initializer)
```

**Comment by Youssef1313** [comment]
Related to #37981 (but doesn't seem to be a dupe)

**Comment by CyrusNajmabadi** [comment]
We would take a targeted community fix here.
